### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by lwindolf** [lwindolf/liferea#974] [Ideas]
Put all your feature requests and ideas here!

Please make one comment per feature and like existing feature comments you want to vote for!

**Comment by horstjens** [comment]
### Synchronization

this may have been asked a million times before, but is there a step-by-step tutorial of how to host the .lifera database on a web server / a google drive so that i can view my feeds synced from my desktop computer as well as from my android mobile devices ?

**Reply by sjehuda** [reply]
### DecSync

Horst, I reply to your comment because it is similar to what I have suggested at #1256, to add DecSync.

> This is a project from @39aldo39 which I want to start a discussion about.
>
> https://github.com/39aldo39/DecSync
> https://github.com/39aldo39/libdecsync

See also #956

**Comment by RafaelLinux** [comment]
### Column Category

Please, it's important to add (or let user enable/disable) the "Category" field on each post, next to "Date" column.

Thank you

**Reply by lwindolf** [reply]
The problem here is that there is no single category per post. Some feeds might have none or just one category, while other have 20.

Having this in a shared column makes the feature unusable :-(

**Reply by sjehuda** [reply]
Perhaps the first category or category that appears the most should appear?

**Comment by johnno66** [comment]
### Caching by time period

 (in Preferences ->Feeds -> Feed Cache Handling)
Please add option to keep items based on age, e.g. only keep items that are <1 week old
This would (in my usage scenario at least) significantly decrease database size.

**Reply by sjehuda** [reply]
> This would (in my usage scenario at least) significantly decrease database size.

This comment is concerning the database size.

I suppose that Liferea compares entry ID and if not available, then it  would find a match of Link or Title, in order to determine whether that entry has been downloaded or not.

In order not to download again an entry, that was read, as a new entry, Liferea has to store a value to identify that entry.

So to make what you want, and I think it's a good idea, Liferea can delete summaries.

---

**UI Consideration**

If Liferea would hide entries:

* No need for UI consideration.

If Liferea would not hide entries:

* Because the rest of the information, like Title, Link and Date, would not be deleted, it might be good to keep those entries visible in the UI.

* This would mean that upon clicking on an item that is more than a week old, Liferea would show no title for it.

* This means that a UI change would have to be made.

* Upon selecting an entry that is older than 7 days, Liferea would display this text "This item is older than 7 days and therefore has been deleted. Click the download button (that button does not exist) to display (download) the item summary".

* * The download process would be inefficient, and even slow, especially when a feed has thousand of entries, because then Liferea would have to (1) Download the feed (2) Lookup for the entry (3) Catch and display the entry summary.

* * Perhaps archiving old summaries in a compressed archive would be a solution to save storage space.

**Comment by payrim** [comment]
### Duplicate subscription

remove duplicate subscription links? i imported from 3 different .opml files and some subscription in between are duplicate but reside in different folders

**Reply by sjehuda** [reply]
payrim, when importing an .opml file, my news bot doesn't import a feed that already appears in its database, but this is because my bot has no support for categories.

Should there be an exception to that if a subscription link appears in a different category? (i.e. a subscription link that integrates both news articles and podcast)

**Reply by lwindolf** [reply]
@payrim I've created issue https://github.com/lwindolf/liferea/issues/1333 to adress this. We actually have the feature already when creating a single subscription, but the check is not triggered for OPML.

**Comment by sjehuda** [comment]
### Select Feeds From Online Database

*This is my attempt to restore a service similar in nature to [Syndic8](https://web.archive.org/web/20011025203909if_/http://www.syndic8.com:80/).*

Provide a functionality to search for feeds from a database, be it from a remote (online) or local (offline) database.

Here is a [database](https://gitgud.io/sjehuda/slixfeed/-/blob/master/slixfeed/assets/feeds.csv?ref_type=heads) I have composed. The database is public domain.

I suggest to make that database accessible via IPFS.
I also suggest to make the practice of a remote feeds database a common practice with any other Feed Reader available.

I think that this is a crucial feature because it would make RSS Readers independent and not subjected to the [whims](https://openrss.org/blog/browsers-should-bring-back-the-rss-button) of web browser vendors.

Thud has this functionality, yet it appears to be using a centralized source to do so. I have opened a similar ticket at https://github.com/aerotoad/Thud/issues/55.

---

**Warning! (Government)**

Because this database is intended to be used with my project, and because it has less than 10K entries, I don't make any selection, which means that some sources _might_ not be appropriate (i.e. nsfw).

Personally, I might have a legal issue with that database at the moment, because I also add sources of neighbouring countries. Some of these sources appear to be censored by the ISP providers here, and this is another reason of why I have an interest of making that database available via a decentralized mean (e.g. IPFS).

**Suggestion to allocate feeds via IPFS**

There would be several databases allocated by language code and country code.

**Unrestricted**
* `de` would include all feeds for use by German speaking audience that is not subjected to restrictions poised by any other German speaking countries (Austria, Germany, Liechtenstein, Luxembourg, Switzerland).

**Restricted**
* `de-at` would include any feed in German, excluding those that are forbidden by Austrian government.
* `de-ch` would include any feed in German, excluding those that are forbidden by Swiss government.
* `de-de` would include any feed in German, excluding those that are forbidden by German government.
* etc.

Unrestricted databases (those that do not have country code), like `de`, will either be unavailable or would be available by enabling an option.

**Reply by lwindolf** [reply]
I thought on how to provide feed catalogues too. I'd actually prefer a PWA mechanism like this that could be shared by any feed reader

- a PWA single page website renders a GUI presenting a catalogue of feeds
- a feed reader embeds the single page app with an internal browser
- the feed reader tells the single page app (via GET param) a URI scheme to use for subscribing
- when a user clicks subscribe the URI scheme of the feed reader is triggered and a subscription happens
- if no URI scheme is provided feed:// is used
- the feed reader can also use custom CSS for the catalogue web app

To allow for some type of federation:

- the PWA is hosted on Github
- user can provide links (as MR) to their forked own catalogues
- the PWA will show user provided links as alternative sources

**Reply by sjehuda** [reply]
@lwindolf do you want me to begin to work on an HTML server-side system to implement this?

I am not yet fully experienced with PWA.

I will make sure that first interface would be usable.

I will use FastAPI, Jinja2 and Uvicorn for server and feedreader as a tool to validate sources.

**Reply by lwindolf** [reply]
@sjehuda The thing about PWAs is you need no server. The browser is the server. I hacked a little prototype with https://github.com/lwindolf/newsagain. It can be used as a PWA, a Cordova Android app and an Cordova Electron desktop app.

I'm currently investigating how to workaround the CORS problem in a suitable way. So giving the user a choice between a browser plugin and a CORS proxy.

**Reply by lwindolf** [reply]
BTW I've outlined my current design goals here https://github.com/lwindolf/newsagain/blob/main/research.md

**Reply by sjehuda** [reply]
I did not have an intention to create a PWA, though I definitely intend to try.

Incidently, I have just received this link from a friend on XMPP https://feedle.world/

I will also create an HTML service based on Python and pywebview, and I will further inform you.

We have several databases of 3K feeds or so.

It will be interesting!

**Reply by sjehuda** [reply]
> BTW I've outlined my current design goals here https://github.com/lwindolf/newsagain/blob/main/research.md

Is the statement at lwindolf/newsagain about replacing Liferea serious? 

Once Liferea would allow to modify its feeds tree with a plugin, prepare to unexpected features that would make Liferea an interesting publishing software.

**Reply by lwindolf** [reply]
@sjehuda Yes, the statement is serious. I'm sorry to disappoint. I believe you that with support for feed list modification a lot of functionality can be build, but I do not want to spend my time on it. I hope you understand this. I wrote about the decision https://lzone.de/liferea/blog/20-years-of-Liferea-and-the-future

**Reply by sjehuda** [reply]
This is a good post!

The concerns of GTK+ were the elements that have deterred me from writing a desktop software in GTK+, albeit that I have desired for it it greately for a long period of time.

The note of libxml2 is good and understandably conveys the message of maintenance concerns.

> This is for example different with another important library for Liferea: libxml2. The API was practically 95% stable for 20 years. Zero maintenance effort!

The interface of Liferea, is a "no nosesnse" interface. It always was beautiful and it will always remain beautiful. It should be the standard.

If the plugin system would have more permissions to manipulate data, I will create several plugins to integrate Liferea with XMPP.

**Comment by sjehuda** [comment]
@lwindolf I ask for your permission to add 3rd-level (i.e. `###`) titles to posts.

Synchronization https://github.com/lwindolf/liferea/discussions/974#discussioncomment-741444

Column Category https://github.com/lwindolf/liferea/discussions/974#discussioncomment-819654

Caching by time period https://github.com/lwindolf/liferea/discussions/974#discussioncomment-1764509

Duplicate subscriptions https://github.com/lwindolf/liferea/discussions/974#discussioncomment-2627646

**Reply by lwindolf** [reply]
@sjehuda I'm sorry I do not understand what you want to do? Do you want to format the linked comments?

**Reply by sjehuda** [reply]
I want to add titles like I did here https://github.com/lwindolf/liferea/discussions/974#discussioncomment-8195898.

It appears that you gave my profile the privilege to do so.

**Reply by lwindolf** [reply]
Feel free to do so :-)

**Reply by sjehuda** [reply]
I added a title to this post https://github.com/lwindolf/liferea/discussions/974#discussioncomment-2627646 for realization

**Reply by sjehuda** [reply]
Thank you

**Comment by luroc** [comment]
### exclude feeds from read/unread

I use Lifera for some time now for following (smaller) blogs and I'm really satisfied, thanks for your work.

I would like to move my general news consumption to Lifera too, but this would destroy the usefulness of the unread counter/view for me. I strive to keep it clean or to only leave articles unread that I actually want to read soon, but with the major news sites there is pretty much always something new.

I would like to have a setting in feeds (or preferably on a folder) to ignore them for the unread counter or alternatively mark all items as read automatically (although still having the read/unread → bold/greyed out styling would be nice). I don't want to have to manually 'mark items read' all the time for busy feeds.

**Reply by lwindolf** [reply]
Such a feature already exists in the feed properties dialog "Advanced" tab: "Mark downloaded items read"

**Reply by luroc** [reply]
I see. As I understand it, it doesn't apply to already downloaded items so I was under the impression that it doesn't have the desired effect and 'download' would be something different in this context, which I couldn't really grasp.

Maybe take this as a UX suggestion to apply the option retroactively to existing items when set.

Also I'd still love if the the bold white vs greyed out styling would remain to see which items I have actually opened, but that's not such a pressing issue.
